Mathlib/Algebra/Ring/Invertible.lean

```diff
@@ -88,6 +88,35 @@ theorem invOf_sub_invOf [Ring R] (a b : R) [Invertible a] [Invertible b] :
     ⅟a - ⅟b = ⅟a * (b - a) * ⅟b := by
   rw [mul_sub, invOf_mul_self, sub_mul, one_mul, mul_assoc, mul_invOf_self, mul_one]
 
+lemma neg_add_eq_mul_invOf_mul_same_iff [Ring R] {a b : R} [Invertible a] [Invertible b] :
+    -(b + a) = a * ⅟b * a ↔ -1 = ⅟a * b + ⅟b * a :=
+  calc -(b + a) = a * ⅟b * a
+      ↔ -a = b + a * ⅟b * a := ⟨by grind, fun h ↦ by simp [h]⟩
+    _ ↔ -a = a * ⅟a * b + a * ⅟b * a := by rw [mul_invOf_self, one_mul]
+    _ ↔ -a = a * (⅟a * b + ⅟b * a) := by simp only [mul_add, mul_assoc]
+    _ ↔ -1 = ⅟a * b + ⅟b * a := ⟨fun h ↦ by simpa using congr_arg (⅟a * ·) h, fun h ↦ by simp [← h]⟩
+
+lemma neg_one_eq_of_invOf_add_eq_add_invOf [Ring R] {a b : R} [Invertible a]
+    [Invertible b] [Invertible (a + b)] (h : ⅟(a + b) = ⅟a + ⅟b) : -1 = ⅟a * b + ⅟b * a := by
```

Comment:
I believe the converse also holds; could you upgrade this to:
```suggestion
lemma neg_one_eq_invOf_mul_add_invOf_mul_iff [Ring R] {a b : R} [Invertible a]
    [Invertible b] [Invertible (a + b)] :
    -1 = ⅟a * b + ⅟b * a ↔ ⅟(a + b) = ⅟a + ⅟b := by
```
and use that below instead?